components/tag-list/tag-list-item-mixin.js

```diff
@@ -131,52 +131,55 @@ export const TagListItemMixin = superclass => class extends LocalizeCoreElement(
 		this.clearable = false;
 		/** @ignore */
 		this.keyboardTooltipItem = false;
-		/** @ignore */
-		this.role = 'listitem';
 		this._displayKeyboardTooltip = false;
 		this._id = getUniqueId();
-		this._tooltipShown = false;
+		this._keyboardTooltipShown = false;
 	}
 
 	firstUpdated(changedProperties) {
 		super.firstUpdated(changedProperties);
 
 		const container = this.shadowRoot.querySelector('.tag-list-item-content');
+
 		this.addEventListener('focus', async(e) => {
 			// ignore focus events coming from inside the tag content
 			if (e.composedPath()[0] !== this) return;
 
-			this._displayKeyboardTooltip = (this.keyboardTooltipItem && !this._tooltipShown);
+			this._displayKeyboardTooltip = (this.keyboardTooltipItem && !this._keyboardTooltipShown);
 			await this.updateComplete;
 
-			/** @ignore */
-			container.dispatchEvent(new FocusEvent('focus', { bubbles: false, cancelable: true }));
+			container.focus();
 		});
+
 		this.addEventListener('blur', () => {
 			this._displayKeyboardTooltip = false;
-			/** @ignore */
-			container.dispatchEvent(new FocusEvent('blur', { bubbles: false, cancelable: true }));
 		});
+
 		this.addEventListener('keydown', this._handleKeydown);
 	}
 
-	handleClearItem(e, clearAll) {
+	_handleClearItem() {
 		if (!this.clearable) return;
 
-		let handleFocus = false;
-		if (e) {
-			const listItemParent = findComposedAncestor(e.composedPath()[0], (node) => node.role === 'listitem');
-			handleFocus = listItemParent ? true : false;
-		}
-
-		if (clearAll) return;
-
 		announce(this.localize('components.tag-list.cleared-item', { value: this.text }));
 
 		/** Dispatched when a user selects to delete an individual tag list item. The consumer must handle the actual element deletion and focus behaviour if there are no remaining list items. */
 		this.dispatchEvent(new CustomEvent(
 			'd2l-tag-list-item-clear',
-			{ bubbles: true, composed: true, detail: { value: this.text, handleFocus } }
+			{ bubbles: true, composed: true, detail: { value: this.text } }
+		));
+	}
+
+	_handleKeyboardTooltipHide() {
```

Comment:
These are just renamed so it's clear that they are just for the keyboard instructions tooltip.

Reply:
This `_displayKeyboardTooltip` could probably be converted to a state property.